src/Parsers/ASTTablesInSelectQuery.cpp

```diff
@@ -112,16 +113,25 @@ void ASTTableExpression::formatImpl(const FormatSettings & settings, FormatState
         settings.ostr << " ";
         database_and_table_name->formatImpl(settings, state, frame);
     }
-    else if (table_function)
+    //In case of table function view, table_function is preferred over subquery for EXPLAIN SYNTAX
+    else if (table_function && table_function->as<ASTFunction>() && table_function->as<ASTFunction>()->name=="view")
```

Comment:
this also does not look well - splitting table function case in two parts, looks inconsistent and like a crutch. May be to avoid this it would be better to just format table function instead of subquery for parametrised view too?

Reply:
For parameterized view to get how the query is actually executed, it would be good to show the subquery. 
Actually I feel for table function view, we can also go with subquery. Please let me know you thoughts

Reply:
I would rather leave first table function check and then subquery check as it was before regardless of the table function type. 

> Actually I feel for table function view, we can also go with subquery. Please let me know you thoughts

This can probably break compatibility.

Reply:
okay, will not break compatibility and keep things same for table function view. 
I have added a parameter in ASTFunction is_parameterized_view, populated by executeTableFunction and will be used here, so for parameterized_view we will display subquery for EXPLAIN SYNTAX.